### PR TITLE
Added Motor Controller Groups + Encoder Reset method

### PR DESCRIPTION
**Pull request by vega0604**
Created RobotMap.java to seclude port numbers for different parts(e.g. encoders)

### CODE REVIEW

File: src/main/java/frc/robot/subsytems/Drivetrain/DriveImpl.java [hunk 1]
```diff
@@ -2,22 +2,30 @@
...
+    Encoder rightEncoder;
 
-    public DriveImpl(SpeedController leftSide, SpeedController rightSide) {
+    public DriveImpl(SpeedControllerGroup leftSide, SpeedControllerGroup rightSide) {
         super(20, TimeUnit.MILLISECONDS);
         this.leftSide = leftSide;
         this.rightSide = rightSide;
         this.drivetrain = new DifferentialDrive(leftSide, rightSide);
+        this.leftEncoder = new Encoder(RobotMap.LEFT_ENCODER_PORTS[0], RobotMap.LEFT_ENCODER_PORTS[1], false, Encoder.EncodingType.k2X);
+        this.rightEncoder = new Encoder(RobotMap.RIGHT_ENCODER_PORTS[0], RobotMap.RIGHT_ENCODER_PORTS[1], false, Encoder.EncodingType.k2X);
```
Comment (andrewazores) [inline src/main/java/frc/robot/subsytems/Drivetrain/DriveImpl.java:28]: Just for the sake of following the pattern, let's pass the Encoders into the constructor the same way that is done with the SpeedControllerGroups. So the Encoders will actually be instantiated in Robot.java.

File: src/main/java/frc/robot/Robot.java [hunk 2]
```diff
@@ -22,7 +25,16 @@
...
   private final SendableChooser<String> m_chooser = new SendableChooser<>();
 
-  private Drive drivetrain;
+  private PWMSparkMax leftMotorController1; 
+  private PWMSparkMax leftMotorController2; 
+  private SpeedControllerGroup leftSide; 
+
+  private PWMSparkMax rightMotorController1;
+  private PWMSparkMax rightMotorController2;
+  private SpeedControllerGroup rightSide;
```
Comment (andrewazores) [inline src/main/java/frc/robot/Robot.java:34]: The Robot.java doesn't need to actually have fields to hold reference to the Sparks and SpeedControllerGroups - those can just be instantiated right when we pass them in to the DriveImpl constructor, either inline or by declaring a local variable within the method where we call DriveImpl() and passing that variable in.

**Review by andrewazores — COMMENTED** [review]
Looking good so far. You could fill in the DriveImpl.distanceDriven() method now too since you've added the Encoders.

**Comment by andrewazores** [comment]
Best/easiest way to respond to PR feedback like this is to simply add more commits to your fork/branch and push them. They'll show up on here as new additions for review. No need to close the PR and re-open a new one.

File: src/main/java/frc/robot/Robot.java [hunk 3]
```diff
@@ -25,15 +26,6 @@
...
-  private PWMSparkMax leftMotorController1; 
-  private PWMSparkMax leftMotorController2; 
-  private SpeedControllerGroup leftSide; 
-
-  private PWMSparkMax rightMotorController1;
-  private PWMSparkMax rightMotorController2;
-  private SpeedControllerGroup rightSide;
-
-
   Drive drivetrain;
```
Comment (bartek717) [inline src/main/java/frc/robot/Robot.java:29]: I think this should be private

Comment (andrewazores) [reply src/main/java/frc/robot/Robot.java:29]: Yea, that would be better practice for good encapsulation.

Comment (vega0604) [reply src/main/java/frc/robot/Robot.java:29]: doesn't allow for accessing in robotSetup for some reason

Comment (andrewazores) [reply src/main/java/frc/robot/Robot.java:29]: That doesn't seem right, it should work/be allowed.

**Comment by bartek717** [comment]
Im going to merge so I can continue to work on it

**Pull request merged**